models/Binance.py

```diff
@@ -5,7 +5,12 @@
 from datetime import datetime, timedelta
 from binance.client import Client
 
-class AuthAPI():
+class AuthAPIBase():
+    def _isMarketValid(self, market):
+        p = re.compile(r"^[A-Z]{6,12}$")
+        return p.match(market)
+
+class AuthAPI(AuthAPIBase):
```

Comment:
Not seen this before. Why the split out of AuthAPIBase rather than include _isMarketValid as part of AuthAPI?

Reply:
Ok, I see. So that both AuthAPI and PublicAPI can both use the function. Correct?

Reply:
Yes like this you can inherit from the AuthAPIBase class and avoid duplicating some logics between classes AuthAPI and PublicAPI.

In Python you can even use inheritance.

In general, I think some refactoring with a helper class for example could really help you to have a more maintenable application.

Reply:
I agree. Thank you for your updates.